Fix boolean response from curl_exec raises php error exception #10

damanic opened the pull request.

woutersamaey commented:
@damanic Why did you make this PR? Which method was giving you problems?

ndeet commented:
@woutersamaey as he describes in the linked #10 issue it seems that if there is an SSL problem or anything else that leads to return `false` and not an actual response there is an exception as soon as we run the first `substr()` on the bool. 

https://github.com/btcpayserver/btcpayserver-greenfield-php/pull/12/files#diff-59d123c8e81abfd2d10704fce897b88c76b7a58de78e86d28d7f47eaa3bd3e78L36

and 

https://github.com/btcpayserver/btcpayserver-greenfield-php/pull/12/files#diff-59d123c8e81abfd2d10704fce897b88c76b7a58de78e86d28d7f47eaa3bd3e78L41

@damanic the formatting is a bit terrible (and our code style CI bot is broken) but I think your fix with still throw an error as you pass `null` as second parameter to the Response constructor? You should initialise `$responseBody` with empty string `''` instead of `null`? 
https://github.com/btcpayserver/btcpayserver-greenfield-php/pull/12/files#diff-59d123c8e81abfd2d10704fce897b88c76b7a58de78e86d28d7f47eaa3bd3e78R63

damanic commented:
Apologies for the code style, my IDE did not pick up on it and I didn't notice until pull request went through.

Refactored to code style. I had to update the  `.php-cs-fixer.dist.php`  to define line endings used, as it was not picking up CRLF / LF difference.

Also addressed the null response issue.  Now if curl_exec returns boolean the client will continue to return a valid response object with status code instead of crashing out.

woutersamaey commented:
I approved the running of the GHA, but they fail.
Also code style still looks very wierd to me. Normally GHA should fix this, right @ndeet ?

Especially the spaces look weird, like in this snippedt:
```
$key                   = $parts[0];
$value                 = $parts[1];
$responseHeaders[$key] = $value;
```

woutersamaey commented:
The idea of the PR is great though. Just wanna see our GHA and formatting work as it should. Guess this is more on us than on you @damanic

damanic commented:
@woutersamaey 
My IDE is set to PSR12 and the code was run through php-cs-fixer.   I did notice that I was aligning consecutive assignments but considered that OK as it passed through the fixer.

Update code style, remove aligned consecutive assignments.

ndeet commented on .php-cs-fixer.dist.php:
```diff
@@ -14,4 +14,4 @@
     'no_unused_imports' => true,
   ]
-)->setFinder($finder);
+)->setLineEnding("\r\n")->setFinder($finder);
```
Interesting that this was needed, did not notice it

ndeet commented:
RE: alignment of assignments, yes seems this is not covered by PSR-12 but something PHPStorm does, maybe we can fix it with .editorconfig or enforce it with cs fixer, but nothing we need to cover in this PR. Thank you @damanic 👍
